src/mono/wasm/debugger/BrowserDebugProxy/EvaluateExpression.cs

```diff
@@ -409,48 +379,25 @@ public static object Evaluate()
                 syntaxTree = findVarNMethodCall.ReplaceVars(syntaxTree, null, null, null, elementAccessValues);
             }
 
-            expressionTree = GetExpressionFromSyntaxTree(syntaxTree);
+            expressionTree = syntaxTree.GetCompilationUnitRoot(token);
             if (expressionTree == null)
                 throw new Exception($"BUG: Unable to evaluate {expression}, could not get expression from the syntax tree");
 
-            MetadataReference[] references = new MetadataReference[]
-            {
-                MetadataReference.CreateFromFile(typeof(object).Assembly.Location),
-                MetadataReference.CreateFromFile(typeof(Enumerable).Assembly.Location)
-            };
-
-            CSharpCompilation compilation = CSharpCompilation.Create(
-                "compileAndRunTheExpression",
-                syntaxTrees: new[] { syntaxTree },
-                references: references,
-                options: new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary));
-
-            SemanticModel semanticModel = compilation.GetSemanticModel(syntaxTree);
-            CodeAnalysis.TypeInfo TypeInfo = semanticModel.GetTypeInfo(expressionTree, cancellationToken: token);
-
-            using (var ms = new MemoryStream())
+            try {
+                var result = await CSharpScript.EvaluateAsync(
+                    string.Join("\n", findVarNMethodCall.variableDefinitions) + "\nreturn " + syntaxTree.ToString(),
+                    ScriptOptions.Default.WithReferences(
+                        typeof(object).Assembly,
+                        typeof(Enumerable).Assembly,
+                        typeof(JObject).Assembly
+                        ),
+                    cancellationToken: token);
```

Comment:
If we specifically catch `CompilationErrorException` here, can we surface the error to VS? or maybe just return the exception message like on line 400.

Reply:
Do I understand it correctly that if the expression itself throws an exception, then we return the message on line 400? Can we detect this specific case of the expression itself throwing, and return a different message?

Reply:
> If we specifically catch `CompilationErrorException` here, can we surface the error to VS? or maybe just return the exception message like on line 400.

IIUC, this would be thrown if our constructed script fails to compile, so maybe the message doesn't need to be shown to the user.